sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/src/ConfigurationManager.cs

```diff
@@ -2,77 +2,207 @@
         {
             serviceInfo = null;
+
             if (serviceId is null)
             {
                 throw new ArgumentNullException(nameof(serviceId));
             }
 
-            if (CallerAppId != null && serviceId.Equals(CallerAppId))
+            if (serviceId.Equals(ConfiguredService.ApplicationId, StringComparison.OrdinalIgnoreCase))
             {
-                serviceInfo = SERVICES.Values.FirstOrDefault(x => x.DefaultService);
-            }
-            else if (SERVICES.ContainsKey(serviceId))
-            {
-                serviceInfo = SERVICES[serviceId];
+                serviceInfo = ConfiguredService;
+                return true;
             }
 
             return serviceInfo != null;
         }
 
-        internal static bool VerifyServiceId(string testId)
+        /// <summary>
+        /// Verify if the service id is valid by checking if we have the service info for it.
+        /// </summary>
+        /// <param name="testId"></param>
+        /// <returns></returns>
+        internal bool VerifyServiceId(string testId)
         {
-            return GetService(testId, out _);
+            return TryGetService(testId, out _);
         }
 
+        /// <summary>
+        /// Get config value from environment variable or use the default value.
+        /// </summary>
+        /// <param name="environmentVariable">Definied Azure function application settings</param>
+        /// <param name="defaultValue">Default value, most likely from auth trigger anotation</param>
+        /// <returns></returns>
         private static string GetConfigValue(string environmentVariable, string defaultValue)
         {
             return Environment.GetEnvironmentVariable(environmentVariable) ?? defaultValue;
         }
 
         private static T GetConfigValue<T>(string environmentVariable, T defaultValue) where T : struct
         {
-            return Environment.GetEnvironmentVariable(environmentVariable) == null ?
+            string value = GetConfigValue(environmentVariable, null);
+
+            return value == null ?
                 defaultValue :
-                (T)Convert.ChangeType(Environment.GetEnvironmentVariable(environmentVariable), typeof(T), CultureInfo.CurrentCulture);
+                (T)Convert.ChangeType(
+                    value: Environment.GetEnvironmentVariable(environmentVariable),
+                    conversionType: typeof(T),
+                    provider: CultureInfo.CurrentCulture);
+        }
+
+        private static ServiceInfo GetAADServiceInfo(string tid)
+        {
+            return new ServiceInfo()
+            {
+                OidcMetadataUrl = "https://login.microsoftonline.com/common/v2.0/.well-known/openid-configuration",
+                TokenIssuerV1 = $"https://sts.windows.net/{tid}/",
+                TokenIssuerV2 = $"https://login.microsoftonline.com/{tid}/v2.0",
+                ApplicationId = "99045fe1-7639-4a75-9d4a-577b6ca3810f",
```

Comment:
Should we add these to a central file that would be easier to locate and update if needed?

Reply:
Or add to the top of the file?

Reply:
Not a central file, it's very related to the confi manager and serviceinfos. For now i would like to leave it in here.